packages/compiler/test/checker/templates.ts

```diff
@@ -81,4 +81,64 @@ describe("compiler: templates", () => {
       character: 15,
     });
   });
+
+  it("allows default template parameters", async () => {
+    testHost.addCadlFile(
+      "main.cadl",
+      `
+        @test model A<T, U = "hi"> { a: T, b: U }
```

Comment:
Is there any stack overflow caused by having circular references in defaults(default self reference). The template had many issues in that area

Reply:
Very good question, will add a test for it (it does work thankfully). But your question made me realize there's another error I need to check for - something like `A<T = U, U = number>` which could work but is pointless and probably an error.